### PR TITLE
service only PR

### PR DESCRIPTION
**Pull request by boazco**

### CODE REVIEW

File: src/main/kotlin/com/hibob/feedback/service/FeedbackService.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    fun getFeedback(feedbackId: UUID, activeUser: ActiveUser): Feedback? {
+        val feedback = feedbackDao.getFeedback(feedbackId, activeUser)
+        if (isNotAuthorized(employeesDao.getEmployeeByActiveUser(activeUser), feedback)) {
+            throw ResponseStatusException(
+                HttpStatus.UNAUTHORIZED,
+                "Unauthorized Access- Trying to fetch feedback which is not yours, while youre not HR or admin"
+            )
+        }
+        return feedback
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/feedback/service/FeedbackService.kt:24]: permission checks should be done in the resource layer

Comment (elijulian) [reply src/main/kotlin/com/hibob/feedback/service/FeedbackService.kt:24]: also, you should be able to use the claims in the token to determine the permissions without needing to make a DB query

Comment (boazco) [reply src/main/kotlin/com/hibob/feedback/service/FeedbackService.kt:24]: so I need to add role and department into the activeUser data class that I receive in the claim?
and how can I check If the activeUser is the feedback writer without the DB?

**Comment by boazco** [comment]
the build is failing because of a problem that I fixed in the next PR

File: src/main/kotlin/com/hibob/feedback/service/FeedbackService.kt (hunk 1 above)
Comment (elijulian) [reply src/main/kotlin/com/hibob/feedback/service/FeedbackService.kt:24]: > so I need to add role and department into the activeUser data class that I receive in the claim?

yes, that would be the best

>  how can I check If the activeUser is the feedback writer without the DB?

you're right that you'll need to fetch the feedback, but you won't need to fetch the employee

Comment (boazco) [reply src/main/kotlin/com/hibob/feedback/service/FeedbackService.kt:24]: from the spec:
API Endpoint: Provide an API endpoint for HR/Admin to view all submitted feedback.

so I dont need anything else only role and department

**Review by elijulian — APPROVED** [review]

**Pull request merged**